cameronRomo/GameStack
# Feature/search and sort games

## Description

This PR adds a search bar for users to fuzzy search games by their names, as well as adds additional sorting options to organize the results by popularity, average rating, release date, name, date added, or relevance. 

The PR also improves the implementation for games with no image URL by adding an image placeholder.

## Related Issue

closes #3 
closes #8 

## Motivation and Context

This change makes searching and sorting games easier by supplying the user with a simple search bar and dropdown input.

## How Has This Been Tested?

This functionality is tested with Cypress by stubbing the request data and testing the response as well as UI changes.

*Note:* Testing all the dropdown options is dificult because the RAWG api does not return all the relative JSON data needed, and with it being a private api, exposing those fields would be challenging and against the "terms of use". Additionally, I expect that that backend functionality has been tested, and I simply want to test that the UI changes according to these requests.

To manually test:

1. Run locally
2. search a game
  * for example search: BioShock
3. Confirm that BioShock games render
4. Select the "Order by: Relevance" dropdown and select one of the options
  * With BioShock previously being searched, selecting "Order by: Release date" should trigger a query to return the list of games in the order they came out
5.  Confirm the UI has changed to display games by the selected sort option

## Types of changes




## Checklist:




- [x] UI changes are responsive.
